microprofile/config/src/main/java/io/helidon/microprofile/config/ConfigCdiExtension.java

```diff
@@ -164,475 +141,271 @@ private void harvestConfigPropertyInjectionPointsFromEnabledBean(@Observes Proce
                 } catch (Exception e) {
                     add.addDeploymentProblem(e);
                 }
             });
         } finally {
             cc.release();
         }
-        ipConfigs.clear();
-
-        LOGGER.exiting(getClass().getName(), "validate");
+        ips.clear();
     }
 
-    private static Class<?> getPropertyClass(ConfigQualifier qualifier) {
-        if (qualifier.rawType().isArray()) {
-            return qualifier.rawType().getComponentType();
-        }
-        if (qualifier.rawType() == qualifier.typeArg()) {
-            return qualifier.rawType();
-        }
-        if (qualifier.typeArg() == qualifier.typeArg2()) {
-            return qualifier.typeArg();
-        }
-
-        return qualifier.typeArg2();
-    }
-
-    @Qualifier
-    @Retention(RUNTIME)
-    @Target({METHOD, FIELD, PARAMETER, TYPE})
-    @interface ConfigQualifier {
-        String fullPath();
-
-        @Nonbinding
-        String key();
-
-        @Nonbinding
-        String defaultValue();
-
-        // e.g. String, Producer, Optional
-        @Nonbinding
-        Class<?> rawType();
-
-        // e.g. eq. to raw type, or type argument of Producer, Optional
-        @Nonbinding
-        Class<?> typeArg();
-
-        @Nonbinding
-        Class<?> typeArg2();
-    }
-
-    static final class ConfigQLiteral extends AnnotationLiteral<ConfigQualifier> implements ConfigQualifier {
-        private String fullPath;
-        private String key;
-        private String defaultValue;
-        private Class<?> rawType;
-        private Class<?> typeArg;
-        private Class<?> typeArg2;
-
-        private ConfigQLiteral(String fullPath,
-                               String key,
-                               String defaultValue,
-                               Class<?> rawType,
-                               Class<?> typeArg,
-                               Class<?> typeArg2) {
-            this.fullPath = fullPath;
-            this.key = key;
-            this.defaultValue = defaultValue;
-            this.rawType = rawType;
-            this.typeArg = typeArg;
-            this.typeArg2 = typeArg2;
-        }
-
-        @Override
-        public String fullPath() {
-            return fullPath;
-        }
-
-        @Override
-        public String key() {
-            return key;
-        }
+    private Object produce(InjectionPoint ip) {
+        ConfigProperty annotation = ip.getAnnotated().getAnnotation(ConfigProperty.class);
+        String injectedName = injectedName(ip);
+        String fullPath = ip.getMember().getDeclaringClass().getName()
+                + "." + injectedName;
+        String configKey = configKey(annotation, fullPath);
 
-        @Override
-        public String defaultValue() {
-            return defaultValue;
+        if (NativeImageHelper.isBuildTime()) {
+            System.err.println("You are accessing configuration key '" + configKey + "' during"
```

Comment:
`System.err` OK?

Reply:
This is only used at native image build time. We cannot rely on logging system, as we may be running in command line using `native-image` or from Maven. `System.err` seemed like the most relevant way to deliver this message to the user.